splitio/producer/storage/split_wrapper.go

```diff
@@ -0,0 +1,153 @@
+package storage
+
+import (
+	"sync"
+
+	"github.com/splitio/go-split-commons/v4/dtos"
+	"github.com/splitio/go-split-commons/v4/storage"
+	"github.com/splitio/go-split-commons/v4/storage/redis"
+	"github.com/splitio/go-toolkit/v5/logging"
+)
+
+// ObservableSplitStorage is an interface extender that adds the method `Count` to the split storage
+type ObservableSplitStorage interface {
+	storage.SplitStorage
+	Count() int
+	Names() []string
+}
+
+// ObservableSplitStorageImpl is an implementaion of the ObservableSplitStorage inteface that wraps an existing storage
+// caches and caches splitnames in-memory (in case the underlying one is non-local, ie: redis)
+type ObservableSplitStorageImpl struct {
+	storage.SplitStorage
+	active *activeSplitTracker
+}
+
+// NewObservableSplitStorage constructs a NewObservableSplitStorage
+func NewObservableSplitStorage(toWrap storage.SplitStorage, logger logging.LoggerInterface) *ObservableSplitStorageImpl {
+
+	names := toWrap.SplitNames()
+	active := newActiveSplitTracker(len(names))
+	active.update(names, nil)
+
+	if _, ok := toWrap.(supportsUpdateWithErrors); !ok {
+		logger.Warning("supplied split storage doesn't report errors, this may introduce inconcistencies in observability endpoints")
+	}
+
+	return &ObservableSplitStorageImpl{
+		SplitStorage: toWrap,
+		active:       active,
+	}
+}
+
+// Update is an override that wraps the original Update method and calls update on the local cache as well
+func (s *ObservableSplitStorageImpl) Update(toAdd []dtos.SplitDTO, toRemove []dtos.SplitDTO, changeNumber int64) {
+	if rs, ok := s.SplitStorage.(supportsUpdateWithErrors); ok {
+		if err := rs.UpdateWithErrors(toAdd, toRemove, changeNumber); err != nil {
+			switch parsedErr := err.(type) {
+			case nil:
+				// no error
+			case *redis.UpdateError:
+				toAdd = filterFailed(toAdd, parsedErr.FailedToAdd)
+				toRemove = filterFailed(toRemove, parsedErr.FailedToRemove)
+			default:
+				// Other types of error are considered critical, meaning nothing got updated,
```

Comment:
Should errors be returned to the caller?

Reply:
no, the original interface doesn't return an error, so we can't do it here either. Eventually when we do commons/v5 we should update all methods to report proper errors